route/build/cpl/nuopc/rof_comp_nuopc.F90

```diff
@@ -750,16 +761,19 @@ subroutine ModelAdvance(gcomp, rc)
     ! Determine if time to write restart
     !--------------------------------
 
-    call ESMF_ClockGetAlarm(clock, alarmname='alarm_restart', alarm=alarm, rc=rc)
-    if (ChkErr(rc,__LINE__,u_FILE_u)) return
-
-    if (ESMF_AlarmIsRinging(alarm, rc=rc)) then
-       if (ChkErr(rc,__LINE__,u_FILE_u)) return
-       rstwr = .true.
-       call ESMF_AlarmRingerOff( alarm, rc=rc )
-       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    rstwr = .false.
+    if (nlend .and. write_restart_at_endofrun) then
+      rstwr = .true.
     else
-       rstwr = .false.
+      call ESMF_ClockGetAlarm(clock, alarmname='alarm_restart', alarm=alarm, rc=rc)
+      if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+      if (ESMF_AlarmIsRinging(alarm, rc=rc)) then
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+         rstwr = .true.
+         call ESMF_AlarmRingerOff( alarm, rc=rc )
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+      endif
```

Comment:
The other thing to change is to move the code to determine if it's time to stop to before the restart alarm.